### PR TITLE
Add support for constants with define()

### LINKED ISSUE #48: Support define() constants
Parsed as a function call node

### PR DESCRIPTION
**Pull request by felixfbecker**
Closes #48

### CODE REVIEW

**Comment by jens1o** [comment]
I think we could run into another problem. https://github.com/felixfbecker/php-language-server/blob/master/src/LanguageServer.php#L283 We would need to watch for every little char, won't we?

**Comment by felixfbecker** [comment]
No, those are just the "explicit" trigger characters.

**Comment by felixfbecker** [comment]
Sure, go ahead! be aware though this has quite some merge conflicts now

**Comment by jens1o** [comment]
So I'm creating a new pr with the tests?

**Comment by felixfbecker** [comment]
Yep

File: src/Protocol/SymbolInformation.php [hunk 1]
```diff
@@ -58,13 +58,25 @@ public static function fromNode(Node $node, string $fqn = null)
...
-        if (!isset($nodeSymbolKindMap[$class])) {
-            throw new Exception("Not a declaration node: $class");
-        }
         $symbol = new self;
-        $symbol->kind = $nodeSymbolKindMap[$class];
-        $symbol->name = (string)$node->name;
+        if (
+            $node instanceof Node\Expr\FuncCall
+            && $node->name instanceof Node\Name
+            && (string)$node->name === 'define'
```
Comment (staabm) [inline src/Protocol/SymbolInformation.php]: Will this false positive on a namespaced function named `define`?

Comment (jens1o) [inline src/Protocol/SymbolInformation.php]: I'll check this, but it should be excluded with `$node->name instanceof Node\Name`

Comment (jens1o) [inline src/Protocol/SymbolInformation.php]: It doesn't care about this, the way I implemented it is right.

**Comment by felixfbecker** [comment]
@jens1o are you still working on this?

**Comment by jens1o** [comment]
It's finished, I just forgot to create the pr. I'll do it now. Thanks for the ping!

**Pull request closed without merging**